deeptime/decomposition/_koopman.py

```diff
@@ -574,23 +574,28 @@ def expectation(self, observables, statistics, lag_multiple=1, observables_mean_
         m_0 = self.mean_0
         m_t = self.mean_t
 
+        integer_multiple = np.issubdtype(type(lag_multiple), np.integer)
+        dtype = float if integer_multiple else complex
         if lag_multiple == 1:
             P = S
         else:
-            p = np.zeros((dim + 1, dim + 1))
+            p = np.zeros((dim + 1, dim + 1), dtype=dtype)
             p[0, 0] = 1.0
             p[1:, 0] = U.T.dot(m_t - m_0)
             p[1:, 1:] = U.T.dot(self.cov_tt).dot(V)
-            P = np.linalg.matrix_power(S.dot(p), lag_multiple - 1).dot(S)
+            if integer_multiple:
+                P = np.linalg.matrix_power(S.dot(p), lag_multiple - 1).dot(S)
+            else:
+                P = (S.dot(p) ** (lag_multiple - 1)).dot(S)
```

Comment:
Is this equivalent to `np.linalg.matrix_power`? Isn't this element-wise power?

Reply:
`np.linalg.matrix_power` only supports integer exponents :slightly_smiling_face: